### PR TITLE
Add missing `gradle/gradle-build-action` setup in `AndroidCIWithGmd`

### PR DESCRIPTION
**Pull request by SimonMarquis**
This will prevent downloading and unzipping Gradle distribution but fetch it from the cache instead like other workflows.

### CODE REVIEW

File: .github/workflows/AndroidCIWithGmd.yaml
Comment (LinX64) [inline .github/workflows/AndroidCIWithGmd.yaml:23]: I'd add a name at the top of the actions we are adding here, the rest of the actions already have names, so let's keep the consistency here as well.

```suggestion
        - name: Setup Gradle
        uses: gradle/gradle-build-action@v2
```

What do you think?

Comment (SimonMarquis) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: Personally I would remove them all, unless they are adding valuable information.

Comment (LinX64) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: Sure, you are free to do whatever you want on your own project, but last time I checked this was supposed to be a project for beginners, so any helpful information can help :]

Comment (SimonMarquis) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: I'll let the maintainers decide what is best here. Not all jobs currently have comments, so if we want to enforce it I'll add them on other entries as well. I'm not against adding comments on self explanatory statements, I only have a strong personal opinion against them 😜

Comment (dturner) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: In code, I'm very much against adding comments to self explanatory statements as it adds noise. For CI, however, there is added benefit in adding a `name` to each step as it makes it clear exactly what that step is doing both in the config and in the job output. E.g. 

image

Suggest we add descriptive names to each step. Can either be done in this PR, or in a separate PR.

**Review by dturner — APPROVED** [review]

File: .github/workflows/AndroidCIWithGmd.yaml
Comment (SimonMarquis) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: Then I guess it would be better to make this a distinct commit, and set names uniformly. I'll do that on top of this PR if that's ok for you :)

Comment (SimonMarquis) [reply .github/workflows/AndroidCIWithGmd.yaml:23]: FYI, the names have been added/restored in #710 with missing steps.

**Comment by abalros** [comment]
gradle/gradle-build-actionالإعداد المفقود بتنسيقAndroidCIWithGmd # 694

**Comment by SimonMarquis** [comment]
@dturner, sorry for the ping 🙏 (could we add [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) in the future?)

**Pull request merged**
